test/rpktest.c

```diff
@@ -0,0 +1,696 @@
+                                          NULL, NULL))
+                || !TEST_true(SSL_set_session(clientssl, client_sess)))
+            goto end;
+
+        /* Set private key (and maybe certificate */
+        if (!TEST_int_eq(SSL_use_PrivateKey_file(serverssl, privkey_file, SSL_FILETYPE_PEM), 1))
+            goto end;
+        if (!TEST_int_eq(SSL_use_certificate_file(serverssl, cert_file, SSL_FILETYPE_PEM), 1))
+            goto end;
+        if (!TEST_int_eq(SSL_check_private_key(serverssl), 1))
+            goto end;
+
+        switch (idx) {
+        default:
+            break;
+        case 11:
+            /* TODO: SSL_add1_expected_peer_rpk(clientssl, client_pkey) */
+            break;
+        case 12:
+            /* TODO: SSL_add1_expected_peer_rpk(clientssl, client_pkey) */
+            SSL_set_options(clientssl, SSL_OP_NO_TICKET);
+            SSL_set_options(serverssl, SSL_OP_NO_TICKET);
+            break;
+        case 13:
+            /* TODO: SSL_add1_expected_peer_rpk(clientssl, client_pkey) */
+            /* TODO: SSL_add1_expected_peer_rpk(serverssl, server_pkey) */
+            /* Use the same key for client auth */
+            if (!TEST_int_eq(SSL_use_PrivateKey_file(clientssl, privkey_file, SSL_FILETYPE_PEM), 1))
+                goto end;
+            if (!TEST_int_eq(SSL_use_certificate_file(clientssl, cert_file, SSL_FILETYPE_PEM), 1))
+                goto end;
+            if (!TEST_int_eq(SSL_check_private_key(clientssl), 1))
+                goto end;
+            SSL_set_verify(serverssl, SSL_VERIFY_PEER | SSL_VERIFY_FAIL_IF_NO_PEER_CERT, rpk_verify_server_cb);
+            break;
+        case 14:
+            /* TODO: SSL_add1_expected_peer_rpk(clientssl, client_pkey) */
+            /* TODO: SSL_add1_expected_peer_rpk(serverssl, server_pkey) */
+            /* Use the same key for client auth */
+            if (!TEST_int_eq(SSL_use_PrivateKey_file(clientssl, privkey_file, SSL_FILETYPE_PEM), 1))
+                goto end;
+            if (!TEST_int_eq(SSL_use_certificate_file(clientssl, cert_file, SSL_FILETYPE_PEM), 1))
+                goto end;
+            if (!TEST_int_eq(SSL_check_private_key(clientssl), 1))
+                goto end;
+            SSL_set_verify(serverssl, SSL_VERIFY_PEER | SSL_VERIFY_FAIL_IF_NO_PEER_CERT, rpk_verify_server_cb);
+            SSL_set_options(serverssl, SSL_OP_NO_TICKET);
+            SSL_set_options(clientssl, SSL_OP_NO_TICKET);
+            break;
+        }
+
+        ret = create_ssl_connection(serverssl, clientssl, SSL_ERROR_NONE);
+        if (!TEST_int_eq(expected, ret))
+            goto end;
+        verify = SSL_get_verify_result(clientssl);
+        if (!TEST_int_eq(client_expected, verify))
+            goto end;
+        if (!TEST_true(SSL_session_reused(clientssl)))
+            goto end;
+
+        /* Get RPK from the session because it may be resumed */
+        if (!TEST_ptr(SSL_get0_peer_rpk(clientssl)))
```

Comment:
The comment above is perplexing. The actual call queries the clientssl handle, where the RPK is part of the session object: https://github.com/openssl/openssl/pull/18185/files#diff-6e1fe8a18fd92c9d48804e59ce4a34f73da70c1e57afb422ed856540fb1f82e7R606

Reply:
In prior iterations, I was wavering between access methods. The comment should be removed.

Reply:
Assuming you'll take care of it.